includes/emails/functions.php

```diff
@@ -376,3 +376,192 @@ function edd_sendwp_disconnect () {
 	wp_send_json_success();
 }
 add_action( 'wp_ajax_edd_sendwp_disconnect', 'edd_sendwp_disconnect' );
+
+/**
+ * Handle installation and activation for Jilt via AJAX
+ *
+ * @since n.n.n
+ */
+function edd_jilt_remote_install_handler() {
+
+	if ( ! current_user_can( 'manage_shop_settings' ) || ! current_user_can( 'install_plugins' ) ) {
+		wp_send_json_error(
+			array(
+				'error' => __( 'You do not have permission to do this.', 'easy-digital-downloads' ),
+			)
+		);
+	}
+
+	include_once ABSPATH . 'wp-admin/includes/plugin-install.php';
+	include_once ABSPATH . 'wp-admin/includes/file.php';
+	include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+
+	$plugins = get_plugins();
+
+	if ( ! array_key_exists( 'jilt-for-edd/jilt-for-edd.php', $plugins ) ) {
+		/*
+		* Use the WordPress Plugins API to get the plugin download link.
+		*/
+		$api = plugins_api(
+			'plugin_information',
+			array(
+				'slug' => 'jilt-for-edd',
+			)
+		);
+
+		if ( is_wp_error( $api ) ) {
+			wp_send_json_error(
+				array(
+					'error' => $api->get_error_message(),
+					'debug' => $api,
+				)
+			);
+		}
+
+		/*
+		* Use the AJAX Upgrader skin to quietly install the plugin.
+		*/
+		$upgrader = new Plugin_Upgrader( new WP_Ajax_Upgrader_Skin() );
+		$install  = $upgrader->install( $api->download_link );
+		if ( is_wp_error( $install ) ) {
+			wp_send_json_error(
+				array(
+					'error' => $install->get_error_message(),
+					'debug' => $api,
+				)
+			);
+		}
+
+		activate_plugin( $upgrader->plugin_info() );
+
+	} else {
+
+		activate_plugin( 'jilt-for-edd/jilt-for-edd.php' );
+	}
+
+	/*
+	* Final check to see if Jilt is available.
+	*/
+	if ( ! class_exists( 'EDD_Jilt_Loader' ) ) {
+		wp_send_json_error(
+			array(
+				'error' => __( 'Something went wrong. Jilt was not installed correctly.', 'easy-digital-downloads' ),
```

Comment:
Noting that I'm getting this error message, although Jilt is actually installed. I think this is related to having 3.0 on a multisite install, though, not an issue with this PR.

Reply:
Can you try reproducing in `master` branch?

Reply:
Ah, thanks, of course I should have done that. Interesting results: have done this several times in multisite with the `master` branch and it's about 50/50 as to whether it installs properly and redirects to Jilt or if the error message happens.

The SendWP install has gone properly in both 2.9 and 3.0.